clang/test/Driver/clang-offload-wrapper-input-multi-bc.cpp

```diff
@@ -0,0 +1,110 @@
+// RUN: echo '%t0.aocx|%t0.prop|%t0.sym'                                     >> %tA.table
+// RUN: echo '[Code|Properties|Symbols]'                                     >  %tB.table
+// RUN: echo '%t1.aocx|%t1.prop|%t1.sym'                                     >> %tB.table
+// RUN: echo '[Code|Properties|Symbols]'                                     >  %tC.table
+// RUN: echo '%t2.aocx|%t2.prop|%t2.sym'                                     >> %tC.table
+y
+// Generate BC files with Properties and Symbols
+// RUN: clang-offload-wrapper "-o=%tA.bc" "-host=x86_64-unknown-linux-gnu" "--emit-reg-funcs=0" "-target=fpga_aocx-intel-unknown" "-kind=sycl" "-batch"  %tA.table
+// RUN: clang-offload-wrapper "-o=%tB.bc" "-host=x86_64-unknown-linux-gnu" "--emit-reg-funcs=0" "-target=fpga_aocx-intel-unknown" "-kind=sycl" "-batch"  %tB.table
+// RUN: clang-offload-wrapper "-o=%tC.bc" "-host=x86_64-unknown-linux-gnu" "--emit-reg-funcs=0" "-target=fpga_aocx-intel-unknown" "-kind=sycl" "-batch"  %tC.table
+
+// Update fake AOCX files
+// RUN: echo 'fake-aocx-0-updated'        > %t0.aocx
+// RUN: echo 'fake-aocx-1-enhanced'       > %t1.aocx
+// RUN: echo 'fake-aocx-2-different-form' > %t2.aocx
+
+// Create BC file with only Code in table
+// RUN: file-table-tform  --extract=Code  %t.table  -o %t_code_only.table
+
+// Create table file that lists symbol and property files
+// RUN: echo "[SymAndProps]" > %t_sym_prop_files.txt
+// RUN: echo %tA.bc >> %t_sym_prop_files.txt
+// RUN: echo %tB.bc >> %t_sym_prop_files.txt
+// RUN: echo %tC.bc >> %t_sym_prop_files.txt
+
+// Generate BC file with only Code coming from the table but everything else coming through --sym-prop-bc-files
+// Thus %t1.bc and %t2.bc should be the same except for their Code.
+// RUN: clang-offload-wrapper "-o=%t2.bc" "-host=x86_64-unknown-linux-gnu" "--emit-reg-funcs=0" "-target=fpga_aocx-intel-unknown" "-kind=sycl" "-batch"  %t_code_only.table --sym-prop-bc-files=%t_sym_prop_files.txt
+
+// RUN: llvm-dis %t1.bc
+// RUN: llvm-dis %t2.bc
+
+// Filter out expected differences
+// ModuleID:                      filename is different
+// sycl_offloading.?.data:        fake AOCX were updated
+// sycl_offloading.device_images: pointer type to offloading data includes the length which is different
+// RUN: grep -v '\(ModuleID\|^@\.sycl_offloading\.[0-9]\.data\|^@\.sycl_offloading\.device_images\)' %t1.ll > %tfiltered1.txt
+// RUN: grep -v '\(ModuleID\|^@\.sycl_offloading\.[0-9]\.data\|^@\.sycl_offloading\.device_images\)' %t2.ll > %tfiltered2.txt
+
+// RUN: cmp %tfiltered1.txt %tfiltered2.txt
+
+// RUN: FileCheck --check-prefix=CHECK_LL1 < %t1.ll %s
+// RUN: FileCheck --check-prefix=CHECK_LL2 < %t2.ll %s
+
+// Check that expected code is found
+// CHECK_LL1: pseudo-aocx-0
+// CHECK_LL1: pseudo-aocx-1
+// CHECK_LL1: pseudo-aocx-2
+
+// CHECK_LL2: fake-aocx-0-updated
+// CHECK_LL2: fake-aocx-1-enhanced
+// CHECK_LL2: fake-aocx-2-different-form
```

Comment:
Does it make sense just to create some prop and table files that are used in this test and put them in the `Inputs/SYCL` directory instead of echoing out here?

Reply:
I could go either way on this.  

Reasons to move to Input/SYCL:
1. Echoing is ugly

Reasons to not move to Input/SYCL:
1. self-contained testcase
2. Currently Inputs/SYCL is for *.o *.bc files that can't be echoed.  Do we want to put new file types in there?

I would prefer having a directory for each test case, but this would also be a new convention.